workflows/python/sdk/order-processor/app.py

```diff
@@ -0,0 +1,81 @@
+import sys
+import threading
+from time import sleep
+from dapr.ext.workflow import WorkflowRuntime, DaprWorkflowClient
+from workflow_activities import order_processing_workflow, notify_activity, process_payment_activity, verify_inventory_activity, update_inventory_activity, requst_approval_activity
+from dapr.clients import DaprClient
+from model import InventoryItem, OrderPayload
+from util import get_address
+
+store_name = "statestore-actors"
+
+default_item_name = "cars"
+
+class WorkflowConsoleApp:    
+    def main(self):
+        print("*** Welcome to the Dapr Workflow console app sample!")
+        print("*** Using this app, you can place orders that start workflows.")
+        print("*** Ensure that Dapr is running in a separate terminal window using the following command:")
+        print("dapr run --dapr-grpc-port 4001 --app-id order-processor")
+        # Wait for the sidecar to become available
+        sleep(5)
+
+        address = get_address()
+        workflowRuntime = WorkflowRuntime()
+        workflowRuntime.register_workflow(order_processing_workflow)
+        workflowRuntime.register_activity(notify_activity)
+        workflowRuntime.register_activity(requst_approval_activity)
+        workflowRuntime.register_activity(verify_inventory_activity)
+        workflowRuntime.register_activity(process_payment_activity)
+        workflowRuntime.register_activity(update_inventory_activity)
+        workflowRuntime.start()
+
+        daprClient = DaprClient(address=f'{address["host"]}:{address["port"]}')
+        baseInventory = {}
+        baseInventory["paperclip"] = InventoryItem("Paperclip", 5, 100)
+        baseInventory["cars"] = InventoryItem("Cars", 15000, 100)
+        baseInventory["computers"] = InventoryItem("Computers", 500, 100)
+
+        self.restock_inventory(daprClient, baseInventory)
+
+        client = DaprWorkflowClient(address["host"], address["port"])
+        sleep(1)
+        print("==========Begin the purchase of item:==========")        
+        item_name = default_item_name
+        order_quantity = 11
+
+        total_cost = int(order_quantity) * baseInventory[item_name].per_item_cost
+        order = OrderPayload(item_name=item_name, quantity=int(order_quantity), total_cost=total_cost)
+        print(f'Starting order workflow, purchasing {order_quantity} of {item_name}')
+        _id = client.schedule_new_workflow(order_processing_workflow, input=order)
+
+        def prompt_for_approval(client: DaprWorkflowClient):
+            client.raise_workflow_event(instance_id=_id, event_name="manager_approval", data={'approval': True})
```

Comment:
This is really cool, but is this interaction testable in our automation?

Reply:
No, it is not testable in our automation. That is why when prompt for approval is called, we directly fire raise_workflow_event, instead of taking some input from user via console.
Actual prompt is mentioned in the fork mentioned here in: https://github.com/dapr/quickstarts/pull/842#issuecomment-1564353362
Let me know if you want to remove this wait/raise external event altogether.

Reply:
Oh i get it now.  Could we add a tiny comment this is for automation?